modin/pandas/groupby.py

```diff
@@ -246,16 +249,18 @@ def _shift(data, periods, freq, axis, fill_value, is_set_nan_rows=True):
             result = self._apply_agg_function(
                 lambda df: df.shift(periods, freq, axis, fill_value)
             )
-            result._query_compiler.set_index_name(None)
+            if self._by is not None:
```

Comment:
Why do we need to set index name to None in this case? A comment should probably be added to the code.

Reply:
comment has been added below

Reply:
added